### DISCUSSION TITLE
Read a message, and how to get the id and other data

### DISCUSSION THREAD

**Discussion by AlkhabazHasan** [ignacio-chiazzo/ruby_whatsapp_sdk#92] [Q&A]
Grate gem and well documented and, it save times of works, but isn't `messages_api.read_message(sender_id: 1234, message_id: "wamid.HBgLMTM0M12345678910=")` needs more explanation? 

It is so easy to get the id for sent message: 
```
message_sent = messages_api.send_text(sender_id: sender, recipient_number: phone_numbers, message: "Hey there! it's Whatsapp Ruby SDK")
message_id = message_sent.data.messages.first.id
puts message_id
```
But for `messages_api.read_message` I have to do it like this:
`params["entry"][0]["changes"][0]["value"]["messages"][0]["id"]`

Is there something I didn't get?
I think I may need some help!

**Comment by ignacio-chiazzo** [comment]
Hi @AlkhabazHasan thanks for reporting. 

Making that change is definitely doable. Note that the Whatsapp API doesn't return the ID of the message, it [just returns](https://developers.facebook.com/docs/whatsapp/cloud-api/guides/mark-message-as-read/) `{ success: true }` if the message is read. 

What we can do, is use the `message_id` send as param in the request. 

If you want to open a PR the changes required are: 
- Add `message_id` as a param in the [ReadMessageResponseData](https://github.com/ignacio-chiazzo/ruby_whatsapp_sdk/blob/078f320447e9ceff27b813ee542b86d7c01bf4b7/lib/whatsapp_sdk/api/responses/read_message_data_response.rb#L15).
- Update the `read_message` method, to pass the `message_id` https://github.com/ignacio-chiazzo/ruby_whatsapp_sdk/blob/main/lib/whatsapp_sdk/api/messages.rb#L439.

**Reply by AlkhabazHasan** [reply]
You are right, it just returns `{ success: true }` if the message was read; that if I have `message_id`. 
My question, Is there another way to get the id rather than like this (for received message) `params["entry"][0]["changes"][0]["value"]["messages"][0]["id"]`?

**Reply by ignacio-chiazzo** [reply]
>, Is there another way to get the id rather than like this (for received message) 

What do you mean by `(for received message)`? Which API? 


Side note: I still think it's a good idea to add `message_id` to the response of `read_message` API. I created an issue for that https://github.com/ignacio-chiazzo/ruby_whatsapp_sdk/issues/93

**Comment by ignacio-chiazzo** [comment]
@AlkhabazHasan there were major updates to the gem that makes it much easier to interact with resposnes. https://github.com/ignacio-chiazzo/ruby_whatsapp_sdk/releases/tag/v1.0.0
